pkg/config/config.go

```diff
@@ -48,6 +64,13 @@ type ComponentSettings struct {
 
 // LocalConfig holds all the config relavent to a specific Component.
 type LocalConfig struct {
+	metav1.TypeMeta   `yaml:",inline"`
+	componentSettings ComponentSettings `yaml:"ComponentSettings,omitempty"`
+}
+
+// ProxyLocalConfig holds all the parameter that local config does but exposes all
+// of it, used for serialization.
+type ProxyLocalConfig struct {
```

Comment:
can we make`ProxyLocalConfig` itself unexported?

Reply:
done